doc/user-guide/testing.rst

```diff
@@ -0,0 +1,246 @@
+.. _testing:
+
+Testing your code
+=================
```

Comment:
The intention of creating this page is that material on using `xarray.testing.assert_equal` etc. should go here, before the material on hypothesis.

Reply:
Perhaps this whole page should go under the xarray internals section of the docs instead of the user guide? Because this realistically is only going to be used by other library developers, not most users.

Reply:
I think we can work this out later. This location seems fine for now, and changing it isn't a backwards-compatibility issue.